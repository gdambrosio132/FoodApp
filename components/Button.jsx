import React from "react";
import { Text, TouchableOpacity } from "react-native";
import { StyleSheet } from "react-native";
function Button({text,onClick}) {
  return (
    <TouchableOpacity onPress={onClick} style={[styles.button]}>
      <Text style = {styles.text}>{text}</Text>
    </TouchableOpacity>
  );
}
export default Button;

const styles = StyleSheet.create({
    button: {
      alignItems: 'center',
      justifyContent: 'center',
      paddingVertical: 12,
      paddingHorizontal: 32,
      borderRadius: 4,
      elevation: 3,
      backgroundColor: 'blue',
    },
    text:{
      color : "white",
      fontSize : 32
    }
  });