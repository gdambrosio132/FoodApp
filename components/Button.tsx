import React from "react";
import { Text, TouchableOpacity } from "react-native";
import { StyleSheet } from "react-native";
function Button({children} : {children:string}) {
  return (
    <TouchableOpacity style={[styles.button]}>
      <Text>Save Calories</Text>
    </TouchableOpacity>
  );
}
export default Button;

const styles = StyleSheet.create({
    button: {
      alignItems: 'center',
      justifyContent: 'center',
      paddingVertical: 12,
      paddingHorizontal: 32,
      borderRadius: 4,
      elevation: 3,
      backgroundColor: 'black',
    },

  });