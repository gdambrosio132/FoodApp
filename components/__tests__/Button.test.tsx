// __tests__
import * as React from "react";
import Button from "../Button";
import renderer from "react-test-renderer";
it(`renders correctly`, () => {
  const tree = renderer.create(<Button>Save Calories</Button>);
  
  expect(tree).toMatchSnapshot();
});