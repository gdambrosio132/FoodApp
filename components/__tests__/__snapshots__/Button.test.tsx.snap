// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "black",
      "borderRadius": 4,
      "elevation": 3,
      "justifyContent": "center",
      "opacity": 1,
      "paddingHorizontal": 32,
      "paddingVertical": 12,
    }
  }
>
  <Text>
    Save Calories
  </Text>
</View>
`;
